plugins/cilium-cni/cilium-cni.go

```diff
@@ -488,18 +488,12 @@ func cmdAdd(args *skel.CmdArgs) (err error) {
 		res.Routes = append(res.Routes, routes...)
 	}
 
-	switch conf.IpamMode {
-	case option.IPAMENI:
+	if conf.IpamMode == option.IPAMENI || conf.IpamMode == option.IPAMAzure {
 		err = eniAdd(ipConfig, ipam.IPV4, conf)
```

Comment:
Please rename this function if we don't exclusively use it for ENI.

Reply:
Done.

Relatedly, most of that function meat recently moved to `pkg/aws/eni/routing` package, which is in no way specific to AWS.

Mind if I rename that to, say, `pkg/nic/routing` for instance (or another name suggestion @tgraf )?

And while at it, also hook in the newly introduced routes deletion on endpoint removal)

Reply:
@bpineau Good point. I think it should be moved into `pkg/datapath/linux/routing`.